File: templates/swift/trie.swift

class Trie {
    
    private var children: [Trie?]
    
    init() {
        children = Array(repeating: nil, count: 2)
    }
    
     func insert(_ num: Int) {
         var root = self
         for i in (0..<32).reversed() {
             let bit = (num >> i) & 1
             if root.children[bit] == nil {
                 root.children[bit] = Trie()
             }
             root = root.children[bit]!
         }
    }
}

File: templates/python/trie.py

class Trie:
    def __init__(self) -> None:
        self.children: list[Trie | None] = [None, None]

    def insert(self, num: int):
        root = self

        for i in range(31, -1, -1):
            bit = (num >> i) & 1
            assert root
            if not root.children[bit]:
                root.children[bit] = Trie()
            root = root.children[bit]
